### DISCUSSION TITLE
Crashworthiness analysis

### DISCUSSION THREAD

**Discussion by Aminofa70** [Ferrite-FEM/Ferrite.jl#1120] [General]
Dear All,

Can Ferrite do Crashworthiness analysis and also apply equivalent static load method for topology optimization ?

Cheers,
Amin

**Comment by KnutAM** [comment]
Hi and welcome @Aminofa70! 

There is no ready-made solution - Ferrite only provides the tools to implement FE code. 

The equivalent static load method (as I understand you simply put the `rho d^2 u/dt^2` term in the right-hand-side) should be straight-forward given that you've done the dynamic simulation. There are also tutorials for how to deal with nonlinear material behavior. For crash, there are currently no tutorials etc. covering how one could do the contact search and formulations.

**Reply by Aminofa70** [reply]
Dear @KnutAM 

Thank you for your reply.
 I am trying to implement it.
 I have already worked on PUPM.jl (topology optimization with Ferrite) and Ferromodo.jl (fusing Ferrite.jl and Comodo.jl). I am also doing Data-Driven FEM using Ferrite that will be published soon. Next door is Dynamic FEM with Ferrite that I am implementing.
I hope with that I won't use any commercial FEM software...

Best,
Amin
